plugins/inputs/ipvs/ipvs.go

```diff
@@ -0,0 +1,85 @@
+// +build linux
+
+package ipvs
+
+import (
+	"errors"
+	"strconv"
+
+	"github.com/amoghe/libipvs"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// IPVS holds the state for this input plugin
+type IPVS struct {
+	handle     libipvs.IPVSHandle
+	handleOpen bool
+}
+
+// Description returns a description string
+func (i *IPVS) Description() string {
+	return "Read IPVS stats"
+}
+
+// SampleConfig returns a sample configuration for this input plugin
+func (i *IPVS) SampleConfig() string {
+	return `
+  ## This plugin takes no configuration, it collects data on all the servers
+  ## (virtual and real) discovered on the system it is running on.
+  ## In the future we may add a way to monitor only a subset of the servers.
+`
+}
+
+// Gather gathers the stats
+func (i *IPVS) Gather(acc telegraf.Accumulator) error {
+
+	// helper: given a Service, return tags that identify it
+	serviceTags := func(s *libipvs.Service) map[string]string {
+		if s.FWMark > 0 {
+			return map[string]string{
+				"fwmark": strconv.Itoa(int(s.FWMark)),
+			}
+		}
+		return map[string]string{
+			"protocol": s.Protocol.String(),
+			"address":  s.Address.String(),
+			"port":     strconv.Itoa(int(s.Port)),
+		}
+	}
+
+	if i.handleOpen == false {
+		h, err := libipvs.New()
```

Comment:
- [X] Done: Disabling load sounds good.
- [X] Done: Will add a note about permissions to the README

Reply:
If the module is loaded before telegraf starts, you may be able to avoid having to run as root if you add these capabilities to the binary:
    cap_net_admin+ep
    cap_net_raw+ep

I haven't tried this with telegraf yet, but it works with collectd's IPVS plugin.

Reply:
Let's verify this and add it to the README.